core_automation/modules/backport_install.py

```diff
@@ -0,0 +1,315 @@
+    env_vars.append(generate_env_var("STRIMZI_CUSTOM_KAFKA_BRIDGE_LABELS", "kafka-bridge", "application"))
+    env_vars.append(generate_env_var("STRIMZI_CUSTOM_KAFKA_EXPORTER_LABELS", "afka-exporter", "application"))
+
+    with open(file_path, 'w') as file:
+        yaml.default_style = '|'
+        yaml.dump(yaml_data, file)
+
+
+def update_deployment(file_path, release_version, subcomp, subcomp_type):
+    with open(file_path, "r") as file:
+        yaml_data = yaml.load(file)
+    common_fields = get_common_fields(yaml_data)
+    containers = yaml_data["spec"]["template"]["spec"]["containers"]
+    labels = yaml_data["spec"]["template"]["metadata"]["labels"]
+    label_env = {
+        "com.company": "Red_Hat",
+        "rht.prod_name": "Red_Hat_Application_Foundations",
+        "rht.prod_ver": format_year_and_quarter(),
+        "rht.comp": "AMQ_Streams",
+        "rht.comp_ver": get_current_streams_version(),
+        "rht.subcomp": subcomp,
+        "rht.subcomp_t": subcomp_type
+    }
+    labels.update(label_env)
+
+    images = {
+        "drain-cleaner": "drain-cleaner-rhel9",
+        "topic-operator": "strimzi-rhel9-operator",
+        "user-operator": "strimzi-rhel9-operator"
+    }
+
+    image_name = images.get(subcomp)
+    if image_name:
+        containers[0]['image'] = f"{common_fields['common_registry_url']}{image_name}:{release_version}"
+
+    if subcomp == "drain-cleaner":
+        containers[0].update({
+            "command": [
+                "/application",
+                "-Dquarkus.http.host=0.0.0.0",
+                "--kafka",
+                "--zookeeper"
+            ]
+        })
+    # Remove the args field
+    if "args" in containers[0]:
+        del containers[0]["args"]
+
+    with open(file_path, 'w') as file:
+        yaml.dump(yaml_data, file)
+
+
+# copy upstream directory to downstream directory
+def copy_directory(source_name, dest_name, strimzi_dir):
+    source_dir = os.path.join(strimzi_dir, source_name)
+    dest_dir = f'../{dest_name}'
+    try:
+        shutil.copytree(source_dir, dest_dir,
+                        dirs_exist_ok=True)  # Copy the contents of the source directory to the destination
+        print(f'{source_name} directory copied to {dest_name} successfully.')
+    except Exception as e:
+        print(f'Error copying {source_name} directory to {dest_name}:', e)
+
+
+def delete_directory(*file_paths):
+    for file_path in file_paths:
+        try:
+            if os.path.isfile(file_path):
+                os.remove(file_path)
+                print(f'Successfully deleted file: {file_path}')
+            elif os.path.isdir(file_path):
+                shutil.rmtree(file_path)
+                print(f'Successfully deleted directory and its contents: {file_path}')
+            else:
+                print(f'File or directory not found: {file_path}')
+        except OSError as e:
+            print(f'Error deleting file or directory {file_path}: {e}')
```

Comment:
Is this function defined in two classes, if so, maybe it is something we can refactor into a shared class